custom_components/ams/__init__.py

```diff
@@ -91,12 +91,12 @@ def read_bytes(self):
 
     def connect(self):
         """Read the data from the port."""
-        if self.meter_manufacturer == "kaifa":
-            parser = Kaifa
-        elif self.meter_manufacturer == "aidon":
-            parser = Aidon
-        else:
-            parser = Kamstrup
+        parser = None
```

Comment:
Could do while self.meter_menufacturer is None and just set the parser as self.meter_manufacturer when found.